panels/models.py

```diff
@@ -10,6 +18,42 @@ def panel_applicants(self):
         return self.query(PanelApplicant).options(joinedload(PanelApplicant.application)).order_by('first_name', 'last_name')
 
 
+class SocialMediaMixin(JSONColumnMixin('social_media', c.SOCIAL_MEDIA)):
+    social_media_urls = config['social_media_urls']
+    social_media_placeholders = config['social_media_placeholders']
+
+    @classmethod
+    def get_placeholder(cls, name):
+        name = cls.unqualify(name)
+        return cls.social_media_placeholders.get(name, '')
+
+    @property
+    def has_social_media(self):
+        return any(getattr(self, f) for f in self.social_media_fields.keys())
+
+    def __getattr__(self, name):
+        if name.endswith('_url'):
+            field_name = self.unqualify(name[:-4])
+            if field_name in self.social_media_fields:
```

Comment:
Where does `social_media_fields` come from...? I see it referenced in multiple places but I don't see where it's defined. I'm sure I'm just missing something, but I figured I'd ask just in case.